nil/internal/network/connection_manager/connection_manager.go

```diff
@@ -0,0 +1,41 @@
+package connection_manager
+
+import (
+	"context"
+
+	libp2pconnmgr "github.com/libp2p/go-libp2p/core/connmgr"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/p2p/net/connmgr"
+	"github.com/rs/zerolog"
+)
+
+type withCustomNotifeeDecorator struct {
+	libp2pconnmgr.ConnManager
+
+	notifee network.Notifiee
+}
+
+func (cm *withCustomNotifeeDecorator) Notifee() network.Notifiee {
+	return cm.notifee
+}
+
+var _ libp2pconnmgr.ConnManager = (*withCustomNotifeeDecorator)(nil)
+
+func NewConnectionManagerWithPeerReputationTracking(
+	ctx context.Context,
+	conf *Config,
+	logger zerolog.Logger,
+	low, hi int,
```

Comment:
I suggest to change the order of arguments here. Usually we pass logger as the last argument.
Here it's impossible to make it the latest but let's move it a bit closer to the end.

Reply:
I don't really want to break up the argument list that is passed to `connmgr.NewConnManager` as is.